llm/gguf.go

```diff
@@ -195,6 +197,30 @@ func (llm *ggufModel) Decode(r io.Reader) error {
 	return nil
 }
 
+func (llm *ggufModel) NumLayers() int {
+	value, exists := llm.kv[fmt.Sprintf("%s.block_count", llm.ModelFamily())]
+	if !exists {
+		return 0
+	}
+
+	// this should be uint32, but handle other types just in case
+	switch v := value.(type) {
+	case uint32:
+		return int(v)
+	case uint64:
+		return int(v)
+	case string:
+		if intValue, err := strconv.Atoi(v); err == nil {
+			return intValue
+		}
+	}
```

Comment:
It's probably fine to check the type assertion since it'll be enforced by the file format